### PR TITLE
refactor: migrate test app package to TypeScript

### PR DESCRIPTION
**Pull request by michelegera**
This PR migrates the `test-app` application to TypeScript and completes the migration started with https://github.com/qonto/ember-prismic-dom/pull/116.

### CODE REVIEW

File: test-app/app/components/group-list-item.hbs [hunk 1]
```diff
@@ -0,0 +1 @@
+<ul>{{yield}} elephant</ul>
```
Comment (vscav) [inline test-app/app/components/group-list-item.hbs:1]:
```suggestion
<ul>{{yield}} elephant</ul>

```

Just to add a new line at the end of file

File: test-app/app/components/list-item.hbs [hunk 2]
```diff
@@ -0,0 +1 @@
+<li>{{yield}} banana</li>
```
Comment (vscav) [inline test-app/app/components/list-item.hbs:1]:
```suggestion
<li>{{yield}} banana</li>

```

Just to add a new line at the end of file

**Comment by vscav** [comment]
We also need to update the CI script to take into account the test-app when running the type-checking with different TS versions. You can have a look on `ember-lottie` for reference:
- [Type checking job in CI script](https://github.com/qonto/ember-lottie/blob/main/.github/workflows/ci.yml#L89-L119)
- [`lint:types` command at the root level](https://github.com/qonto/ember-lottie/blob/main/package.json#L15)

File: test-app/app/components/group-list-item.hbs (hunk 1 above)
Comment (michelegera) [reply test-app/app/components/group-list-item.hbs:1]: I think this is prevented by `.editoconfig`:

```toml
[*.hbs]
insert_final_newline = false
```

Do you think we should flip this setting?

Comment (vscav) [reply test-app/app/components/group-list-item.hbs:1]: Ah my bad! No that's fine, I just think our addons are not consistent regarding this config. You can ignore my other comments regarding this 👌

**Comment by michelegera** [comment]
@vscav Thanks for the review. [This commit](https://github.com/qonto/ember-prismic-dom/pull/132/commits/2413ab03c4040aa090cc4446eea823165fdb02e2) adds type-checking with different TS versions for `test-app` to the CI script, while the `lint:types` command was already there.

I’ve also added a comment to explain why we use `any` for `Hyperlink`’s `node` argument.

File: test-app/app/components/hyperlink.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import templateOnlyComponent from '@ember/component/template-only';
+
+export interface HyperlinkSignature {
+  Element: HTMLAnchorElement;
+  // The `node` argument should be a TreeNode, but that type is not exported
+  // from the package.
```
Comment (vscav) [inline test-app/app/components/hyperlink.ts:6]: 🙏

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -136,5 +136,8 @@ jobs:
       - name: Update TS version
```
Comment (vscav) [inline .github/workflows/ci.yml:136]: Just to be fully aligned with the other addons:

```suggestion
      - name: Update TS version on addon package
```

**Review by vscav — APPROVED** [review]
Thanks @michelegera 🙏 🚀

**Pull request merged**